components/UnifiedDocFeed/FeedMenu/FeedMenu.tsx

```diff
@@ -0,0 +1,236 @@
+import { css, StyleSheet } from "aphrodite";
+import { breakpoints } from "~/config/themes/screen";
+import { useState, useEffect, useMemo, useRef } from "react";
+import colors, { pillNavColors } from "~/config/themes/colors";
+import FeedMenuSortDropdown from "./FeedMenuSortDropdown";
+import { feedTypeOpts } from "../constants/UnifiedDocFilters";
+import { useRouter } from "next/router";
+import { connect } from "react-redux";
+import { getSelectedUrlFilters } from "../utils/getSelectedUrlFilters";
+import handleFilterSelect from "../utils/handleFilterSelect";
+import FeedMenuTab from "./FeedMenuTab";
+import FeedMenuTopLevelFilters from "./FeedMenuTopLevelFilters";
+import useEffectForOutsideMenuClick from "../utils/useEffectForOutsideMenuClick";
+import FeedMenuMobileScrollControls from "./FeedMenuMobileScrollControls";
+
+type Args = {
+  hubState?: any;
+};
+
+const FeedMenu = ({ hubState }: Args) => {
+  const router = useRouter();
+  const tabsContainerRef = useRef<HTMLInputElement | null>(null);
+  const [viewportWidth, setViewportWidth] = useState(0);
+  const [tagsMenuOpenFor, setTagsMenuOpenFor] = useState(null);
+  const isHomeOrMyHubs = ["/", "/my-hubs"].includes(router.pathname);
+  const selectedFilters = useMemo(() => {
+    return getSelectedUrlFilters({
+      query: router.query,
+      pathname: router.pathname,
+    });
+  }, [router.pathname, router.query]);
+
+  useEffect(() => {
+    useEffectForOutsideMenuClick({
+      setTagsMenuOpenFor,
+    });
+  }, []);
+
+  useEffect(() => {
+    const _setViewportWidth = () => setViewportWidth(window.innerWidth);
+
+    window.addEventListener("resize", _setViewportWidth, true);
+
+    return () => {
+      window.removeEventListener("resize", _setViewportWidth, true);
+    };
+  }, []);
+
+  const getTabFilters = ({ selectedFilters }) => {
+    const _renderOption = (opt) => {
+      return (
+        <div className={css(styles.labelContainer)}>
+          <span className={css(styles.iconWrapper)}>{opt.icon}</span>
+          <span>{opt.label}</span>
+        </div>
+      );
+    };
+
+    const tabs = Object.values(feedTypeOpts).map((opt) => ({
+      html: _renderOption(opt),
+      ...opt,
+    }));
+
+    const tabsAsHTML = tabs.map((tabObj) => {
+      if (tabObj.value === selectedFilters.type) {
+        // @ts-ignore
+        tabObj.isSelected = true;
+      }
+      return tabObj;
+    });
+
+    return tabsAsHTML;
+  };
+
+  const tabElems = useMemo(() => {
+    const tabs = getTabFilters({ selectedFilters });
+    return tabs.map((t) => (
+      <FeedMenuTab
+        selectedFilters={selectedFilters}
+        tabObj={t}
+        handleOpenTagsMenu={(forType) => setTagsMenuOpenFor(forType)}
+        handleFilterSelect={(selected) =>
+          handleFilterSelect({ router, ...selected })
+        }
+        isTagsMenuOpen={tagsMenuOpenFor === t.value}
+      />
+    ));
```

Comment:
Unnecessary variable assignment
```suggestion
    return getTabFilters({ selectedFilters }).map((t) => (
      <FeedMenuTab
        selectedFilters={selectedFilters}
        tabObj={t}
        handleOpenTagsMenu={(forType) => setTagsMenuOpenFor(forType)}
        handleFilterSelect={(selected) =>
          handleFilterSelect({ router, ...selected })
        }
        isTagsMenuOpen={tagsMenuOpenFor === t.value}
      />
    ));
```

Reply:
Good call